controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -1195,6 +1199,156 @@ def test_controller_get_capabilities(self):
+
+    def _prepare_expand_volume_mocks(self, storage_agent):
+        storage_agent.return_value = self.storage_agent
+
+        self.mediator.expand_volume = Mock()
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_zero_size(self, storage_agent):
+        self._prepare_expand_volume_mocks(storage_agent)
+
+        self.request.capacity_range.required_bytes = 0
+        self.servicer.ControllerExpandVolume(self.request, self.context)
+        self.assertEqual(self.context.code, grpc.StatusCode.OK)
+        self.mediator.expand_volume.assert_called_once_with(volume_id=self.volume_id, required_bytes=0)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_big_size(self, storage_agent):
+        self._prepare_expand_volume_mocks(storage_agent)
+
+        self.request.capacity_range.required_bytes = 1125899906842625
+        self.servicer.ControllerExpandVolume(self.request, self.context)
+        self.assertEqual(self.context.code, grpc.StatusCode.OUT_OF_RANGE)
+        self.mediator.expand_volume.assert_not_called()
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_succeeds_mount_access_mode(self, storage_agent):
+        self._prepare_expand_volume_mocks(storage_agent)
+
+        expand_respond = self.servicer.ControllerExpandVolume(self.request, self.context)
+        self.assertEqual(self.context.code, grpc.StatusCode.OK)
+        self.assertTrue(expand_respond.node_expansion_required)
+        self.mediator.expand_volume.assert_called_once_with(volume_id=self.volume_id,
+                                                            required_bytes=self.capacity_bytes)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_succeeds_block_access_mode(self, storage_agent):
+        self._prepare_expand_volume_mocks(storage_agent)
+
+        self.request.volume_capability = csi_pb2.VolumeCapability(
+            access_mode=csi_pb2.VolumeCapability.AccessMode(mode=self.access_types.SINGLE_NODE_WRITER),
+            block=csi_pb2.VolumeCapability.BlockVolume())
+
+        expand_respond = self.servicer.ControllerExpandVolume(self.request, self.context)
+        self.assertEqual(self.context.code, grpc.StatusCode.OK)
+        self.assertTrue(expand_respond.node_expansion_required)
+        self.mediator.expand_volume.assert_called_once_with(volume_id=self.volume_id,
+                                                            required_bytes=self.capacity_bytes)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_wrong_secrets(self, a_enter):
+        self._test_create_object_with_wrong_secrets(a_enter)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_wrong_volume_capabilities(self, storage_agent):

```

Comment:
split this into two unit tests (something like `test_expand_volume_with_invalid_fstype_fail` and `test_expand_volume_with_invalid_access_mode_fail`)

Reply:
ok
done